snuba/utils/streams/processing.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC, abstractmethod
+from typing import (
+    Callable,
+    Generic,
+    Mapping,
+    Optional,
+    Sequence,
+    Type,
+)
+
+from snuba.utils.streams.consumer import Consumer, ConsumerError
+from snuba.utils.streams.types import Message, Partition, Topic, TPayload
+
+
+logger = logging.getLogger(__name__)
+
+
+class ProcessorFactory(ABC, Generic[TPayload]):
+    @abstractmethod
+    def create(
+        self, commit: Callable[[Mapping[Partition, int]], None]
+    ) -> Processor[TPayload]:
+        raise NotImplementedError
+
+
+class Processor(ABC, Generic[TPayload]):
```

Comment:
What about calling this `ProcessingStrategy` (or `StreamProcessingStrategy` or `ProcessorStrategy`) so that it captures the relationship between this class and StreamProcessor. They are fairly coupled, so I think it would make sense to give this class a name that explains which responsibility it has in the context of a StreamProcessor (thus strategy, as the Strategy design pattern defines it).

Also having to deal with Processor and StreamProcessor (where StreamProcessor is not a Processor) seems confusing.

Reply:
I think this makes sense, will update.